api/src/main/java/org/openmrs/module/locationbasedaccess/LocationBasedAccessActivator.java

```diff
@@ -45,6 +47,19 @@ public void willStart() {
      * @see ModuleActivator#started()
      */
     public void started() {
+        String refAppLocationGlobalProperty = Context.getAdministrationService().getGlobalProperty(LocationBasedAccessConstants.REF_APP_LOCATION_USER_PROPERTY_NAME);
+        if(StringUtils.isBlank(refAppLocationGlobalProperty)) {
+            Context.getAdministrationService().setGlobalProperty(LocationBasedAccessConstants.REF_APP_LOCATION_USER_PROPERTY_NAME,
+                    LocationBasedAccessConstants.LOCATION_USER_PROPERTY_NAME);
+            if(log.isDebugEnabled()) {
+                log.debug("RefApp Location global property created with value - " + LocationBasedAccessConstants.LOCATION_USER_PROPERTY_NAME);
+            }
+        }
+        else if(StringUtils.isNotBlank(refAppLocationGlobalProperty) && !refAppLocationGlobalProperty.equals(LocationBasedAccessConstants.LOCATION_USER_PROPERTY_NAME)) {
```

Comment:
I thought this was just a flag that is set to either true or false?

Reply:
Then we need two Global properties for this implementation. 
1. One for True or False (to show locations or not)
2. To get the location user property name
Do want me to make this by two GPs?

Reply:
What happens when it points to wrong property name?

Reply:
that might be a problem. So better to remove this and, follow Whenever it is set, we know that it is true. If not set, we know it is false.

Reply:
In this line, it is checking for the exisiting value with the actual value to show the WARN message.